Automattic/wp-calypso
# Automatically create a default shipping zone with Free Shipping if the site doesn't have any configured

Related: #13612

In this PR:
* Added logic to the "Setup tasks" screen to check if Shipping is configured. Shipping is considered "configured" if there is at least 1 shipping method created **and** enabled in all of the shipping zones.
* When opening the "Shipping" settings screen, if there are no shipping zones, a default one for the base country is created, with a Free Shipping method configured for everyone.

To test:
* The store must be in an "Egg" state, that is, you should see the "setup tasks list".
* Go to WP-Admin and remove all the shipping zones. If the Rest of the World zone has any method configured, remove it.
* Go to the Calypso WooCommerce Dashboard. You should see the "Shipping" option unchecked.
* Go to the Shipping settings page. When the shipping zones load, you should see that a zone for the configured country (`US` or `Canada`) was added, and inside it a `Free Shipping` method was configured.
* Go back to the Dashboard, see that the "Shipping" task now is checked.
* Go to the Shipping settings page and disable the `Free Shipping` method, so that there are no enabled shipping methods configured.
* Go back to the Dashboard, check that the `Shipping` task is unchecked again.

Caveats:
* If you remove all the shipping zones and go back to the Shipping Settings screen, the country zone & Free Shipping method is re-created. This must be just a 1-time thing, I'm working on fixing that but it shouldn't be a blocker.
